File: experiments/04_register_model/run_model_register.py
from azureml.core import Environment, Experiment, Workspace, ScriptRunConfig
from azureml.core.model import Model

ws = Workspace.from_config(path="../../")

env = Environment.from_conda_specification(
  name="sklearn-iris",
  file_path="./conda.yml"
)

config = ScriptRunConfig(
  source_directory="./src",
  script="train.py",
  compute_target="rimi-cluster",
  environment=env,
  arguments=[
    "--kernel", "linear",
    "--penalty", 2.0,
  ]
)

exp = Experiment(
  workspace=ws,
  name="iris-model-register",
)

run = exp.submit(config=config)
run.wait_for_completion(show_output=True)

# run.download_file(
#   name="outputs/model.joblib",
#   output_file_path="./model/model.joblib"
# )
# 
# model = Model.register(
#   workspace=ws,
#   model_path="./model/model.joblib",
#   model_name="iris-sklearn",
# )

model = run.register_model(
  model_name="iris-sklearn",
  model_path="outputs/model.joblib",
)

File: r/deployments/01_aci/run_model_register.R
library(azuremlsdk)

ws <- load_workspace_from_config(path = "../../../")

exp <- experiment(
  workspace = ws,
  name = "accidents-logistic-regression"
)
env <- r_environment(
  name = "r-environment",
  custom_docker_image = "9054a1a822d04d21a7a5f4e726da20cc.azurecr.io/r-azureml:latest" # nolint
)

config <- estimator(
  source_directory = "./src",
  entry_script = "train.R",
  compute_target = "rimi-cluster",
  environment = env
)

run <- submit_experiment(
  experiment = exp,
  config = config
)

wait_for_run_completion(run, show_output = TRUE)

register_model_from_run(
  run = run,
  model_name = "accidents-logistic",
  model_path = "outputs/model.rds"
)
